com.unity.render-pipelines.universal/Runtime/Decal/DBuffer/DBufferRenderPass.cs

```diff
@@ -52,72 +48,64 @@ public DBufferRenderPass(Material dBufferClear, DBufferSettings settings, DecalD
+                var desc = cameraData.cameraTargetDescriptor;
                 desc.graphicsFormat = QualitySettings.activeColorSpace == ColorSpace.Linear ? GraphicsFormat.R8G8B8A8_SRGB : GraphicsFormat.R8G8B8A8_UNorm;
                 desc.depthBufferBits = 0;
                 desc.msaaSamples = 1;
 
-                cmd.GetTemporaryRT(Shader.PropertyToID(s_DBufferNames[0]), desc);
+                RenderingUtils.ReAllocateIfNeeded(ref dBufferColorHandles[0], desc, name: s_DBufferNames[0]);
             }
 
             if (m_Settings.surfaceData == DecalSurfaceData.AlbedoNormal || m_Settings.surfaceData == DecalSurfaceData.AlbedoNormalMAOS)
             {
-                var desc = renderingData.cameraData.cameraTargetDescriptor;
+                var desc = cameraData.cameraTargetDescriptor;
                 desc.graphicsFormat = GraphicsFormat.R8G8B8A8_UNorm;
                 desc.depthBufferBits = 0;
                 desc.msaaSamples = 1;
 
-                cmd.GetTemporaryRT(Shader.PropertyToID(s_DBufferNames[1]), desc);
+                RenderingUtils.ReAllocateIfNeeded(ref dBufferColorHandles[1], desc, name: s_DBufferNames[1]);
             }
 
             if (m_Settings.surfaceData == DecalSurfaceData.AlbedoNormalMAOS)
             {
-                var desc = renderingData.cameraData.cameraTargetDescriptor;
+                var desc = cameraData.cameraTargetDescriptor;
                 desc.graphicsFormat = GraphicsFormat.R8G8B8A8_UNorm;
                 desc.depthBufferBits = 0;
                 desc.msaaSamples = 1;
 
-                cmd.GetTemporaryRT(Shader.PropertyToID(s_DBufferNames[2]), desc);
+                RenderingUtils.ReAllocateIfNeeded(ref dBufferColorHandles[2], desc, name: s_DBufferNames[2]);
             }
 
             // depth
             RTHandle depthHandle;
-            if (!isDeferred)
+            if (isDeferred)
+            {
+                depthHandle = cameraData.renderer.cameraDepthTargetHandle;
+            }
+            else
             {
-                var depthDesc = renderingData.cameraData.cameraTargetDescriptor;
+                var depthDesc = cameraData.cameraTargetDescriptor;
                 depthDesc.graphicsFormat = GraphicsFormat.None; //Depth only rendering
-                depthDesc.depthStencilFormat = renderingData.cameraData.cameraTargetDescriptor.depthStencilFormat;
+                depthDesc.depthStencilFormat = cameraData.cameraTargetDescriptor.depthStencilFormat;
                 depthDesc.msaaSamples = 1;
 
-                cmd.GetTemporaryRT(Shader.PropertyToID(m_DBufferDepth.name), depthDesc);
+                RenderingUtils.ReAllocateIfNeeded(ref m_DBufferDepth, depthDesc, name: s_DBufferDepthName);
                 depthHandle = m_DBufferDepth;
             }
-            else
-            {
-                depthHandle = m_CameraDepthAttachment;
-            }
 
             ConfigureTarget(dBufferColorHandles, depthHandle);
```

Comment:
I noticed we usually call ConfigureTarget on camera setup, wouldn't be issue calling it outside it?

Reply:
I'm not sure. So far, I don't think so.
The bigger issues I had was getting targets from universal renderer too early.

Reply:
K so if it works it is okay. I think I will split it in my https://github.com/Unity-Technologies/Graphics/pull/6442/files.